docs/core/tools/global-tools-how-to-create.md

```diff
@@ -0,0 +1,226 @@
+
+First, create a new .NET Core 2.1 Console Application.
+
+```console
+dotnet new console -o botsay
+```
+
+Enter the `botsay` directory created by the previous command.
+
+In this example we'll create a console application that generates an ascii bot and prints a message. The first thing we'll do is use the same command line parsing system that the .NET Core CLI tool uses.
+
+Add reference to the `Microsoft.Extensions.CommandLineUtils` NuGet package.
+
+```console
+dotnet add package Microsoft.Extensions.CommandLineUtils
+```
+
+## Add the code
+
+Open the `Program.cs` file.
+
+The first thing we'll code is support for the command line options the program will use. Add the following `using` directives to the top of the file. This will include the members of the command line package we referenced from NuGet.
+
+```csharp
+using System.Reflection;
+using Microsoft.Extensions.CommandLineUtils;
+```
+
+Next, move down to the `static void Main(string[] args)` method. Most likely you have a print statement in there. Erase any code inside this method.
+
+Add the following code to process the command line arguments for your application. The `Microsoft.Extensions.CommandLineUtils` NuGet package makes it simple to support command line switches, flags, and it also provides a help system.
+
+```csharp
+// Create the command line host
+var app = new CommandLineApplication();
+
+// Set the basics of the command line
+app.Name = "botsay";
+app.Description = "Prints an ascii bot that says something in your terminal.";
+app.VersionOption("-v | --version", Assembly.GetEntryAssembly().GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion);
+app.HelpOption("-h | --help");
+
+// Add our own command line argument: message
+var messageArg = app.Argument("<message>", "The message to print.");
+
+// Parse the command line.
+app.Execute(args);
+```
+
+The `CommandLineApplication` object defines all of the command line arguments your application uses. Besides the built in `help` and `version` argument options, the code above adds a new argument named `message` that will store the message the user wants to print.
+
+Next, add code to check if the `message` argument was properly passed, and if not, display the help information and quit.
+
+```csharp
+// Check if a message was provided
+if (string.IsNullOrEmpty(messageArg.Value))
+{
+    if (!app.IsShowingInformation)
+        app.ShowHelp();
+
+    return;
+}
+```
+
+If the check above succeeds, then the message argument was not passed and the application will quit. To help the user know that they need to provide the `message` argument, `app.ShowHelp()` is called. However, the `app.IsShowingInformation` will be set to `true` when the `help` or `version` argument options are passed, so we don't want to display the help information twice. 
+
+### Create the bot
```

Comment:
Somewhere mention it's the .NET Bot or dotnetbot as I expect it is copyrighted. 

(It's not just any bot after all :)